analytical_engine/core/object/dynamic.cc

```diff
@@ -0,0 +1,59 @@
+/** Copyright 2020 Alibaba Group Holding Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifdef NETWORKX
+
+#include <numeric>
+#include <string>
+
+#include "core/object/dynamic.h"
+
+/* static definition */
+gs::dynamic::AllocatorT gs::dynamic::Value::allocator_;
+
+std::size_t gs::dynamic::Value::hash() const {
+  switch (GetType()) {
+  case rapidjson::kNullType:
+    return 0xBAAAAAAD;
+  case rapidjson::kObjectType: {
+    return 0xBAAAAAAE;
```

Comment:
If all node that has oid type `kObjectType` (i.e., json object), the hash value would be the same and they will be routed into the same fragment.

That sounds not the desired behaviour.

Reply:
Change to not allow object as node( like NetworkX do), if the value is object, throw an runtime error.